### PR TITLE
VMware vCenter proxy + storage policies states & dependencies

### PR DESCRIPTION
**Pull request by alexbleotu**
### What does this PR do?

Added a vCenter specific proxy and the functions required to configure storage policies (previously known as storage profiles) on the vCenter. This requires to connect to a different endpoint called `/pbm` which names the state and utils modules. The execution function is `vsphere`

Examples of the input expected to configure a storage policy can be found in the sysdoc of `salt.states.pbm`.

List of functions:
Proxy:
- all required functions to run the `vcenter` proxy
- some existing functions and tests in the `vsphere` module have been adjusted to support the new proxy

Utils functions
- get_new_service_instance_stub that creates a new service instance stub
- salt.utils.pbm.get_profile_manager
- salt.utils.pbm.get_placement_solver
- salt.utils.pbm.get_capability_definitions
- salt.utils.pbm.get_policies_by_id
- salt.utils.pbm.get_storage_policies
- salt.utils.pbm.create_storage_policy
- salt.utils.pbm.create_storage_policy
- salt.utils.pbm.update_storage_policy
- salt.utils.pbm.get_default_storage_policy_of_datastore
- salt.utils.pbm.assign_default_storage_policy_to_datastore

Execution functions:

- salt.modules.vsphere._get_policy_dict that transforms a policy VMware object into a dict representation
- salt.modules.vsphere.list_storage_policies that retrieves dict representations of storage policies, filtered by name
- salt.modules.vsphere.list_default_vsan_policy that retrieves dict representation of the default storage policies
- salt.modules.vsphere._get_capability_definition_dict that transforms a VMware capability definition into a dict representation
- salt.modules.vsphere.list_capability_definitions that returns dict representations of VMware capability definition
- salt.modules.vsphere._apply_policy_config that applies a storage dict representations values to a object
- salt.modules.vsphere.create_storage_policy
- salt.modules.vsphere.update_storage_policy
- salt.modules.vsphere.list_default_storage_policy_of_datastore that lists the dict representation of the policy assigned by default to a datastore
- salt.modules.vsphere.assign_default_storage_policy_to_datastore

States functions:

- salt.states.pbm.default_vsan_policy_configured state that configures the default storage policy
- salt.states.pbm.storage_policies_configured state that creates/configures storage policies
- salt.states.pbm.default_storage_policy_assigned state that manages default storage policies to datastore assigments

### Tests written?

Yes (for utils functions)

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by alexbleotu** [comment]
The state functions use some shared methods in https://github.com/saltstack/salt/pull/43645 so I that needs to be merged in first

**Comment by cachedout** [comment]
@alexbleotu I got #43645 in but there are some merge conflicts now. Could you please resolve them? Thanks.

**Comment by alexbleotu** [comment]
@cachedout rebased and added the pyVmomi incompatibility check

**Review by rallytime — APPROVED** [review]
Very well done!

**Comment by alexbleotu** [comment]
Can we get this merged in? I really need to get both this PR and https://github.com/saltstack/salt/pull/43751 (afterwards please) merged in today or tomorrow. Thanks!

**Comment by rallytime** [comment]
re-run py3

File: salt/utils/vmware.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # -*- coding: utf-8 -*-
 '''
+import sys
```
Comment (cachedout) [inline salt/utils/vmware.py]: Import is in the comment block.

File: salt/utils/vmware.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
...
+import ssl
```
Comment (cachedout) [inline salt/utils/vmware.py]: Import is in the comment block

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/utils/vmware.py (hunk 2 above)
Comment (alexbleotu) [reply salt/utils/vmware.py]: my bad will take them out

File: salt/utils/vmware.py (hunk 1 above)
Comment (alexbleotu) [reply salt/utils/vmware.py]: Done

File: salt/utils/vmware.py (hunk 2 above)
Comment (alexbleotu) [reply salt/utils/vmware.py]: Done

**Comment by alexbleotu** [comment]
@cachedout  all done. Is the GPG required now?

**Comment by rallytime** [comment]
@alexbleotu We're going to be encouraging contributors to sign their commits, but you don't need to do so retroactively.

**Pull request merged**